core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -88,89 +85,125 @@ public ClientLogger(String className) {
     }
 
     /**
-     * Sets the logger to the verbose logging level.
+     * Logs a format-able message that uses {@code {}} as the placeholder at {@code verbose} log level
      *
-     * @return Updated ClientLogger if verbose is enabled.
-     */
-    public ClientLogger asVerbose() {
-        return asLevel(VERBOSE_LEVEL);
-    }
-
-    /**
-     * Sets the logger to the info logging level.
+     * <p><strong>Code Samples</strong></p>
+     * <p>
+     * Logging a message at verbose log level
+     * <pre>
+     * ClientLogger logger = new ClientLogger(Example.class);
+     * logger.logAsVerbose("A format-able message. Hello, {}", name);
+     * </pre>
      *
-     * @return Updated ClientLogger if info is enabled.
+     * @param format The format-able message to log
+     * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
-    public ClientLogger asInfo() {
-        return asLevel(INFORMATIONAL_LEVEL);
+    public void logAsVerbose(String format, Object... args) {
```

Comment:
in terms of naming, if we take this approach I would probably say we should call it `verbose` and remove the `logAs` prefix from all methods.

Reply:
Okay. Does this have to be consistent with other languages on what the method name should be?

Reply:
No, we don't need consistency across languages